### ISSUE TITLE
Migrate imports from PCH file to every file.

### ISSUE THREAD

**Issue by st3fan** [Leanplum/Leanplum-iOS-SDK#28]
(Mozilla / Firefox)

When dropping the Leanplum source code directly in a project, the code does not compile unless we include a number of `#import <Foundation/Foundation.h>` and `#import <UIKit/UIKit.h>` statements.

See the attached diff.

This is a little odd, it seems the main culprit here is `__IPHONE_OS_VERSION_MIN_REQUIRED` being undefined. Why is that? Our project has the *Minimal Deployment Target* set to `9.3` - so I was expecting that to do the right thing for that check.

```
diff --git a/Leanplum-SDK/Classes/Constants.h b/Leanplum-SDK/Classes/Constants.h
index e1c6abe..0c09bfb 100644
--- a/Leanplum-SDK/Classes/Constants.h
+++ b/Leanplum-SDK/Classes/Constants.h
@@ -23,6 +23,7 @@
 //  under the License.

 #import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>

 #define IOS_6_SUPPORTED defined(_ARM_ARCH_7) || defined(__i386__) || defined(__LP64__)

diff --git a/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Categories/NSDate+RFC1123.m b/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Categories/NSDate+RFC1123.m
index e874a88..c17310e 100755
--- a/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Categories/NSDate+RFC1123.m
+++ b/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Categories/NSDate+RFC1123.m
@@ -6,6 +6,7 @@
 //  http://blog.mro.name/2009/08/nsdateformatter-http-header/
 //
 //  No obvious license attached
+#import <Foundation/Foundation.h>
 #if __IPHONE_OS_VERSION_MIN_REQUIRED < 70000

 #import "NSDate+RFC1123.h"
diff --git a/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Categories/NSDictionary+RequestEncoding.m b/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Categories/NSDictionary+RequestEncoding.m
index cc26e48..18c3735 100755
--- a/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Categories/NSDictionary+RequestEncoding.m
+++ b/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Categories/NSDictionary+RequestEncoding.m
@@ -22,6 +22,9 @@
 //  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 //  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 //  THE SOFTWARE.
+
+#import <Foundation/Foundation.h>
+
 #if __IPHONE_OS_VERSION_MIN_REQUIRED < 70000

 #import "MKNetworkKit.h"
diff --git a/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Categories/NSString+MKNetworkKitAdditions.m b/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Categories/NSString+MKNetworkKitAdditions.m
index f11b193..6ed66f0 100755
--- a/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Categories/NSString+MKNetworkKitAdditions.m
+++ b/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Categories/NSString+MKNetworkKitAdditions.m
@@ -22,6 +22,7 @@
 //  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 //  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 //  THE SOFTWARE.
+#import <Foundation/Foundation.h>
 #if __IPHONE_OS_VERSION_MIN_REQUIRED < 70000

 #import "NSString+MKNetworkKitAdditions.h"
diff --git a/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNKEngineWrapper.m b/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNKEngineWrapper.m
index 2b24be6..2f86b09 100644
--- a/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNKEngineWrapper.m
+++ b/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNKEngineWrapper.m
@@ -5,6 +5,7 @@
 //  Created by Alexis Oyama on 11/14/16.
 //  Copyright (c) 2016 Leanplum, Inc. All rights reserved.
 //
+#import <Foundation/Foundation.h>
 #if __IPHONE_OS_VERSION_MIN_REQUIRED < 70000

 #import "Leanplum_MKNKEngineWrapper.h"
diff --git a/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNKOperationWrapper.m b/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNKOperationWrapper.m
index 4e11bb8..14b10a1 100644
--- a/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNKOperationWrapper.m
+++ b/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNKOperationWrapper.m
@@ -5,6 +5,7 @@
 //  Created by Alexis Oyama on 11/14/16.
 //  Copyright (c) 2016 Leanplum, Inc. All rights reserved.
 //
+#import <Foundation/Foundation.h>
 #if __IPHONE_OS_VERSION_MIN_REQUIRED < 70000

 #import "Leanplum_MKNKOperationWrapper.h"
diff --git a/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNetworkEngine.m b/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNetworkEngine.m
index a35d86a..749dda2 100755
--- a/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNetworkEngine.m
+++ b/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNetworkEngine.m
@@ -22,6 +22,7 @@
 //  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 //  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 //  THE SOFTWARE.
+#import <Foundation/Foundation.h>
 #if __IPHONE_OS_VERSION_MIN_REQUIRED < 70000

 #import "MKNetworkKit.h"
diff --git a/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNetworkOperation.m b/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNetworkOperation.m
index f5b4e18..17a0e68 100755
--- a/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNetworkOperation.m
+++ b/Leanplum-SDK/Classes/Vendor/MKNetworkKit/Leanplum_MKNetworkOperation.m
@@ -22,6 +22,9 @@
 //  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 //  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 //  THE SOFTWARE.
+
+#import <Foundation/Foundation.h>
+
 #if __IPHONE_OS_VERSION_MIN_REQUIRED < 70000

 #import "MKNetworkKit.h"
diff --git a/Leanplum-SDK/Classes/Vendor/Reachability/Leanplum_Reachability.m b/Leanplum-SDK/Classes/Vendor/Reachability/Leanplum_Reachability.m
index 97559ba..0ac31c9 100755
--- a/Leanplum-SDK/Classes/Vendor/Reachability/Leanplum_Reachability.m
+++ b/Leanplum-SDK/Classes/Vendor/Reachability/Leanplum_Reachability.m
@@ -25,6 +25,7 @@ CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
  POSSIBILITY OF SUCH DAMAGE.
  */

+#import <Foundation/Foundation.h>
 #import "Leanplum_Reachability.h"


diff --git a/Leanplum-SDK/Classes/Vendor/UniqueIdentifier/UIDevice+IdentifierAddition.m b/Leanplum-SDK/Classes/Vendor/UniqueIdentifier/UIDevice+IdentifierAddition.m
index 44612b7..12f769b 100755
--- a/Leanplum-SDK/Classes/Vendor/UniqueIdentifier/UIDevice+IdentifierAddition.m
+++ b/Leanplum-SDK/Classes/Vendor/UniqueIdentifier/UIDevice+IdentifierAddition.m
@@ -6,6 +6,9 @@
 //  Copyright 2011 Aurora Apps. All rights reserved.
 //

+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+
 #import "UIDevice+IdentifierAddition.h"
 #import "NSString+MD5Addition.h"
```

**Comment by benmarten** [comment]
dropping the source code into your project is not recommended at this point, since we use cocoa pods.

you can include the source code like this:
in your apps Podfile, point to the location of the Leanplum-iOS-SDK repo, e.g.: 
```
    pod 'Leanplum-iOS-SDK-sources', :path => '../Leanplum-iOS-SDK'
```

**Comment by milos1290** [comment]
You can also use Carthage as dependency manager. As for dropping the code, we rely on .pch file which has added Foundation and UIKit .

**Comment by st3fan** [comment]
@milos1290 The reason I am doing this is because if I want to work on a patch for the SDK, I need the actual source code in my project.

What kind of workflow does Leanplum use to work on the SDK?

**Comment by st3fan** [comment]
i'm curious why you are using a `.pch` and not just including the proper headers.

Can I convince you to include the includes or imports back in the sources? When using precompiled headers they will be ignored, so it should not make a difference. It would make working with this code a lot simpler.

**Comment by milos1290** [comment]
@st3fan We use cocoa pods internally for development. Basically the example project contains sample app, running `pod install` will configure sample app and add SDK as development pod, which contains all sources instead of built framework. As for `.pch` file, i think it was legacy, we should definitely remove it and add needed headers to our source files instead of relying on it.

Also, if you want to add SDK to your project as source to develop just add `pod 'Leanplum-iOS-SDK', :path => 'path/to/our/sdk'` in your podfile,

**Comment by milos1290** [comment]
fixed in #30, closing.
